src/oemof/solph/flows/_non_convex_invest_flow.py

```diff
@@ -0,0 +1,917 @@
+        nonconvex = NonConvex(
+            startup_costs=startup_costs,
+            shutdown_costs=shutdown_costs,
+            activity_costs=activity_costs,
+            inactivity_costs=inactivity_costs,
+            minimum_uptime=minimum_uptime,
+            minimum_downtime=minimum_downtime,
+            maximum_startups=maximum_startups,
+            maximum_shutdowns=maximum_shutdowns,
+            initial_status=initial_status,
+            positive_gradient=positive_gradient,
+            negative_gradient=negative_gradient,
+        )
+        super().__init__(nonconvex=nonconvex, **kwargs)
+
+
+class NonConvexInvestFlowBlock(SimpleBlock):
+    r"""
+    **The following sets are created:** (-> see basic sets at
+    :class:`.Model` )
+
+    A set of flows with the attribute `nonconvex` of type
+        :class:`.options.NonConvex`.
+    MIN_FLOWS
+        A subset of set NONCONVEX_FLOWS with the attribute `min`
+        being not None in the first timestep.
+    ACTIVITYCOSTFLOWS
+        A subset of set NONCONVEX_FLOWS with the attribute
+        `activity_costs` being not None.
+    INACTIVITYCOSTFLOWS
+        A subset of set NONCONVEX_FLOWS with the attribute
+        `inactivity_costs` being not None.
+    STARTUPFLOWS
+        A subset of set NONCONVEX_FLOWS with the attribute
+        `maximum_startups` or `startup_costs`
+        being not None.
+    MAXSTARTUPFLOWS
+        A subset of set STARTUPFLOWS with the attribute
+        `maximum_startups` being not None.
+    SHUTDOWNFLOWS
+        A subset of set NONCONVEX_FLOWS with the attribute
+        `maximum_shutdowns` or `shutdown_costs`
+        being not None.
+    MAXSHUTDOWNFLOWS
+        A subset of set SHUTDOWNFLOWS with the attribute
+        `maximum_shutdowns` being not None.
+    MINUPTIMEFLOWS
+        A subset of set NONCONVEX_FLOWS with the attribute
+        `minimum_uptime` being not None.
+    MINDOWNTIMEFLOWS
+        A subset of set NONCONVEX_FLOWS with the attribute
+        `minimum_downtime` being not None.
+    POSITIVE_GRADIENT_FLOWS
+        A subset of set NONCONVEX_FLOWS with the attribute
+        `positive_gradient` being not None.
+    NEGATIVE_GRADIENT_FLOWS
+        A subset of set NONCONVEX_FLOWS with the attribute
+        `negative_gradient` being not None.
+
+
+    **Variables**
+
+    * :math:`P(t)`
+        Actual flow value (created in :class:`oemof.solph.models.BaseModel`).
+
+    * :math:`P_{invest}`
+        Value of the investment variable, i.e. equivalent to the nominal
+        value of the flows after optimization.
+
+    Status variable (binary) `om.NonConvexInvestmentFlowBlock.status`:
+        Variable indicating if flow is >= 0 indexed by FLOWS
+
+    * :math: `new_param(i,o,t)` (non-negative real number)
+        new paramater representing the multiplication of
+        `P_{invest}` and status(i,o,t)`used for the constraints
+        on the minimum and maximum flow constraints.
```

Comment:
What about `P_{status}`?

Reply:
> What about `P_{status}`?

I think `P_{status}` is NOT required here or anywhere else in this flow class. As long as the `P_{invest}` is larger than zero, `P_{status}` is always `True`, and when `P_{invest}` becomes zero, it means `P_{status}` would also be equal to zero. On the other side, by removing this integer variable from the problem, the computation speed will be faster while results won't change.

Reply:
I wanted to suggest to (re)name the parameter `new_param` (to) `P_{status}`. As it is the product of the nominal power and the status, the name would make sense, imho.